carapace-server/src/main/java/org/carapaceproxy/configstore/HerdDBConfigurationStore.java

```diff
@@ -376,4 +416,48 @@ public void saveCertificate(CertificateData cert) {
         }
     }
 
+    @Override
+    public void saveAcmeChallengeToken(String id, String data) {
+        try (Connection con = datasource.getConnection();
+                PreparedStatement psInsert = con.prepareStatement(INSERT_INTO_ACME_CHALLENGE_TOKENS_TABLE)) {
+            psInsert.setString(1, id);
+            psInsert.setString(2, data);
+            psInsert.executeUpdate();
+        } catch (Exception err) {
+            LOG.log(Level.SEVERE, "Error while performing saving of ACME challenge token with id: " + id + " data: " + data, err);
+            throw new ConfigurationStoreException(err);
+        }
+    }
+
+    @Override
+    public String loadAcmeChallengeToken(String id) {
+        try (Connection con = datasource.getConnection()) {
+            try (PreparedStatement ps = con.prepareStatement(SELECT_FROM_ACME_CHALLENGE_TOKENS_TABLE)) {
+                ps.setString(1, id);
+                try (ResultSet rs = ps.executeQuery()) {
+                    if (rs.next()) {
+                        return rs.getString(2);
```

Comment:
why returning the 2nd element ? you can narrow the query an return only that column